### PR TITLE
[EP-2410] Branded checkout improvements

### PR DESCRIPTION
**Pull request by canac**
Changes:

* Log error when tokenization or encryption fails
* Scroll to the section of the branded checkout form that matches the "Change" button the user clicked
* Send review_order event after add_payment_info

https://jira.cru.org/browse/EP-2410

### CODE REVIEW

File: src/app/branded/branded-checkout.component.js [hunk 1]
```diff
@@ -78,17 +79,43 @@ class BrandedCheckoutController {
...
+          break
+        case 'payment':
+          scrollElement = 'checkout-existing-payment-methods'
+          break
+      }
     }
-    this.$window.document.querySelector('branded-checkout').scrollIntoView({ behavior: 'smooth' })
+
+    // Watch for changes until the element we are scrolling to exists and everything has loaded
+    // because there will be layout shift every time a new component finishes loading
```
Comment (wrandall22) [inline src/app/branded/branded-checkout.component.js:106]: Hmm, my experience has always been that the loading indicator remains until all of the pieces have finished loading. Did you run into other cases in your testing?

File: src/app/branded/branded-checkout.component.js [hunk 2]
```diff
@@ -78,17 +79,43 @@ class BrandedCheckoutController {
...
+    const observer = new this.$window.MutationObserver(() => {
+      // TODO: When Firefox supports :has(), this query can be changed to `.panel :has(${scrollElement})`
```
Comment (wrandall22) [inline src/app/branded/branded-checkout.component.js:108]: Note that give site must support older versions of browsers also, so we need to be careful using anything that is not supported by like 99% of browsers (we had to support IE 8 up until 2016, for example).

Comment (canac) [reply src/app/branded/branded-checkout.component.js]: Good point. I updated the comment.

**Review by wrandall22 — APPROVED** [review]

**Comment by canac** [comment]
@wrandall22 FYI I'm doing some more testing in staging before I merge this and might have a few more tweaks.

**Comment by canac** [comment]
Tested in staging ✅

**Review by wrandall22 — APPROVED** [review]

**Pull request merged**